stl/inc/ranges

```diff
@@ -12,6 +12,7 @@
 _EMIT_STL_WARNING(STL4038, "The contents of <ranges> are available only with C++20 or later.");
 #else // ^^^ !defined(__cpp_lib_ranges) / defined(__cpp_lib_ranges) vvv
 #include <__msvc_int128.hpp>
+#include <algorithm>

```

Comment:
There is no need to introduce a huge `<algorithm>`.

Reply:
In fact, we now use `_RANGES max`, and thus need `<algorithm>`

Reply:
I suspect we need `std::max`, and even if we need it, we shouldn't introduce `<algorithm>`. We can move the `max` definition to the small header and introduce it. There is no need to introduce `<algorithm>` just for `std::max`.

Reply:
I did initially implement the parts of `zip_view` which require `max` with separate redundant algorithms which didn't make use of it in the first place, thus avoiding the `<algorithm>` header, but it would be better to make use of the existing STL functions. Besides, as you mentioned, having these in `<xutility>` will make it easier to use them in the implementation of other types. I'll make that change here. (I would do that in a separate PR, but I was foolish enough to base this PR on the `main` branch of my fork. Some lessons are learned the hard way, I suppose.)

Quick question, however: For reference, what is the MSVC STL's stance on breaking transitive includes? I know that in this particular case, it shouldn't break anything (since `<algorithm>` already includes `<xutility>`), but there might be other instances where this might happen. I would expect that the current treatment is to avoid breaking them when possible, but allowing them if necessary, since the STL user technically shouldn't be relying on them, anyways.

Reply:
> what is the MSVC STL's stance on breaking transitive includes?

We do this from time to time, but there has to be a decent reason (like a significant throughput gain), because dealing with the source-breaking impact is a headache. For example, I most recently did this in #2996 and #2998, and even with the escape hatches added by #3027, it broke almost a dozen real-world projects.